tls/datadog_checks/tls/tls.py

```diff
@@ -334,38 +330,3 @@ def local_cert_loader(self, cert):
         if b'-----BEGIN CERTIFICATE-----' in cert:
             return load_pem_x509_certificate(cert, backend)
         return load_der_x509_certificate(cert, backend)
-
-    @property
-    def tls_context(self):
-        if self._tls_context is None:
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext
-            # https://docs.python.org/3/library/ssl.html#ssl.PROTOCOL_TLS
-            self._tls_context = ssl.SSLContext(protocol=PROTOCOL_TLS_CLIENT)
-
-            # Run our own validation later on if need be
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.check_hostname
-            #
-            # IMPORTANT: This must be set before verify_mode in Python 3.7+, see:
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.check_hostname
-            self._tls_context.check_hostname = False
-
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.verify_mode
-            self._tls_context.verify_mode = ssl.CERT_REQUIRED if self._validate_cert else ssl.CERT_NONE
-
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_verify_locations
-            if self._cafile or self._capath:  # no cov
-                self._tls_context.load_verify_locations(self._cafile, self._capath, None)
-
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_default_certs
-            else:
-                self._tls_context.load_default_certs(ssl.Purpose.SERVER_AUTH)
-
-            # https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_cert_chain
-            if self._cert:  # no cov
-                self._tls_context.load_cert_chain(self._cert, keyfile=self._private_key)
-
-            # https://docs.python.org/3/library/ssl.html#ssl.create_default_context
-            if 'SSLv3' in self._allowed_versions:  # no cov
-                self._tls_context.options &= ~ssl.OP_NO_SSLv3
-
-        return self._tls_context
```

Comment:
Currently seeing how to ensure the logic of creating this context while also using updated SSL context